### PR TITLE
NRF53: fix broken rptun

### PR DESCRIPTION
**Pull request by raiden00pl**
## Summary

- cmake: port 99b0bad94e arch/armv8-m: DSP extension is optional
- boards/nrf53: add missing shmem section
- arch/nrf53: move SPU configuration to a separate file
- arch/nrf53/rptun: remove dependency on DEV_SIMPLE_ADDRENV
- boards/nrf5340-dk: enable rpmsg uart for rptun configurations
- arch/nrf53: FLASH cache only for netcore, disable by default as it breaks rptu
    After sevaral resets of the chip after programming, rptun stops working correctly.
    Simple test with RPMSG UART causes a lock:
    
    on the netcore:
        `nsh>cat /dev/ttyproxy`
    
    on the appcore:
        `nsh>echo xxx > /dev/ttyproxy`
    
    The problem doesn't occur with NRF53_FLASH_PREFETCH=n
    
## Testing
nrf5340-dk with RPMSG UART

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: boards/arm/nrf53/thingy53/scripts/flash_app.ld [hunk 1]
```diff
@@ -30,6 +30,7 @@ MEMORY
 {
     progmem (rx)  : ORIGIN = 0x00000000, LENGTH = 1024K
     datamem (rwx) : ORIGIN = 0x20000000, LENGTH = 512K - RPTUN_SHMEM
+    shmem (rw) : ORIGIN = 0x20078000, LENGTH = RPTUN_SHMEM
```
Comment (pkarashchenko) [inline boards/arm/nrf53/thingy53/scripts/flash_app.ld:33]:
```suggestion
    progmem (rx)  : ORIGIN = 0x00000000, LENGTH = 1024K
    datamem (rwx) : ORIGIN = 0x20000000, LENGTH = 512K - RPTUN_SHMEM
    shmem (rw)    : ORIGIN = 0x20078000, LENGTH = RPTUN_SHMEM
```

File: boards/arm/nrf53/nrf5340-dk/scripts/flash_app.ld [hunk 2]
```diff
@@ -30,6 +30,7 @@ MEMORY
 {
     progmem (rx)  : ORIGIN = 0x00000000, LENGTH = 1024K
     datamem (rwx) : ORIGIN = 0x20000000, LENGTH = 512K - RPTUN_SHMEM
+    shmem (rw) : ORIGIN = 0x20078000, LENGTH = RPTUN_SHMEM
```
Comment (pkarashchenko) [inline boards/arm/nrf53/nrf5340-dk/scripts/flash_app.ld:33]:
```suggestion
    progmem (rx)  : ORIGIN = 0x00000000, LENGTH = 1024K
    datamem (rwx) : ORIGIN = 0x20000000, LENGTH = 512K - RPTUN_SHMEM
    shmem (rw)    : ORIGIN = 0x20078000, LENGTH = RPTUN_SHMEM
```

File: boards/arm/nrf53/nrf5340-audio-dk/scripts/flash_app.ld [hunk 3]
```diff
@@ -30,6 +30,7 @@ MEMORY
 {
     progmem (rx)  : ORIGIN = 0x00000000, LENGTH = 1024K
     datamem (rwx) : ORIGIN = 0x20000000, LENGTH = 512K - RPTUN_SHMEM
+    shmem (rw) : ORIGIN = 0x20078000, LENGTH = RPTUN_SHMEM
```
Comment (pkarashchenko) [inline boards/arm/nrf53/nrf5340-audio-dk/scripts/flash_app.ld:33]:
```suggestion
    progmem (rx)  : ORIGIN = 0x00000000, LENGTH = 1024K
    datamem (rwx) : ORIGIN = 0x20000000, LENGTH = 512K - RPTUN_SHMEM
    shmem (rw)    : ORIGIN = 0x20078000, LENGTH = RPTUN_SHMEM
```

File: boards/arm/nrf53/common/scripts/flash_app.ld [hunk 4]
```diff
@@ -53,6 +53,7 @@ MEMORY
                     LENGTH = FLASH_IMG_SIZE
     datamem (rwx) : ORIGIN = RAM_CPUAPP_START_ADDR,
                     LENGTH = CONFIG_NRF53_CPUAPP_MEM_RAM_SIZE - RPTUN_SHMEM
+    shmem (rw) : ORIGIN = 0x20078000, LENGTH = RPTUN_SHMEM
```
Comment (pkarashchenko) [inline boards/arm/nrf53/common/scripts/flash_app.ld:56]:
```suggestion
    datamem (rwx) : ORIGIN = RAM_CPUAPP_START_ADDR,
                    LENGTH = CONFIG_NRF53_CPUAPP_MEM_RAM_SIZE - RPTUN_SHMEM
    shmem (rw)    : ORIGIN = 0x20078000, LENGTH = RPTUN_SHMEM
```

File: arch/arm/src/nrf53/nrf53_spu.h [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+#include <nuttx/config.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* SPU only for secure app core image */
+
+#if defined(CONFIG_NRF53_APPCORE) &&            \
```
Comment (pkarashchenko) [inline arch/arm/src/nrf53/nrf53_spu.h:36]:
```suggestion
#if defined(CONFIG_NRF53_APPCORE) && \
```

File: boards/arm/nrf53/thingy53/scripts/flash_app.ld (hunk 1 above)
Comment (raiden00pl) [reply boards/arm/nrf53/thingy53/scripts/flash_app.ld]: done

**Review by jerpelea — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**